src/twisted/web/_newclient.py

```diff
@@ -929,6 +928,7 @@ def writeTo(self, transport):
                 self._writeToEmptyBodyContentLength(transport)
             else:
                 self._writeHeaders(transport, None)
+            return succeed(None)
         elif self.bodyProducer.length is UNKNOWN_LENGTH:
```

Comment:
Can we conver this to an `if` ?

Not sure if there is a performance implicatin, but it should make the code easier to read.


```suggestion

        if self.bodyProducer.length is UNKNOWN_LENGTH:
```

Reply:
I tried it, I think it makes the code a little less clear.